motr/io_nw_xfer.c

```diff
@@ -528,15 +528,29 @@ static void target_ioreq_seg_add(struct target_ioreq              *ti,
 
 	/* For checksum of Parity Unit the global object offset will be
 	 * assumed to be aligned with PG start offset, so removing the
-	 * additional value NxUS w.r.t PG start, which is added by the 
+	 * additional value NxUS w.r.t PG start, which is added by the
 	 * nw_xfer_io_distribute() function. src.sa_unit = layout_n(play) + unit
-	 * Removing this offset N will help to compute PG unit idx as 0,1..,k-1 
+	 * Removing this offset N will help to compute PG unit idx as 0,1..,k-1
 	 * which is the index of pi_paritybufs
-	 */ 
-	goff_cksum = unit_type == M0_PUT_DATA ? gob_offset : 
+	 */
+	goff_cksum = unit_type == M0_PUT_DATA ? gob_offset :
 		     (gob_offset + (src->sa_unit - layout_n(play)) *
 		     layout_unit_size(play));
-	
+
+	/**
+	 * There are scenarios where K > N in such case when the
+	 * unit type is M0_PUT_PARITY, adding [(K-N) x gob offset] into
+	 * global object offset so that right PG and Unit index will get
+	 * computed.
+	 */
+	if ((unit_type == M0_PUT_PARITY) &&
+	   (layout_k(play) > layout_n(play))) {
+		m0_bcount_t goff_delta = (layout_k(play) -
+					  layout_n(play)) *
+					  gob_offset;
+		goff_cksum += goff_delta;
```

Comment:
I didn't see this algorithm in the doc:
https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1010663429/Motr+DI?src=mail&src.mail.action=view&src.mail.notification=com.atlassian.confluence.plugins.confluence-notifications-batch-plugin%3Abatching-notification&src.mail.recipient=8a7f808a7812f8e60178188e03000549&src.mail.timestamp=1652423407033

Maybe we need to add a paragraph there to explain such special case.

Reply:
sure, this particular case was a miss and was captured little late during check in time; RMW ST with configuration of sync replication didnot capture this which was tested heavily for DI; thats one part to additionally check in that rmw ST :)